### PR TITLE
Dialogs layout alignment

### PR DESCRIPTION
**Pull request by RussKie**
Relates to #6183

## Proposed changes
* Add a new base dialog control `GitExtensionsDialog`that features two main sections - `MainPanel` for the main content, and `ControlPanel` for all buttons. All form should ideally inherit from this dialog to provide a consistent look and feel.
* Update `FormInit` layout
* Update `FormClone` layout
* Update `FormStatus` layout
* Update `FormProgress` layout
* Update `FormRemoteProgress` layout

⚠️ This work concentrates only on the placement of buttons, the original layout of dialogs is preserved (for the most part), and changes to the actual layout of each individual form is outside the scope of this work.

### Before

* `FormInit`
    image

* `FormClone`
    image

* `FormClone`
    image

### After

* `FormInit`
    image
    image

* `FormClone`
    image
    image

* `FormClone`
    ge-status-dark
    ge-status-light

## Test environment(s) 

- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4180.0
- DPI 144dpi (150% scaling)

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/CommandsDialogs/FormInit.cs [hunk 1]
```diff
@@ -26,9 +26,14 @@ public partial class FormInit : GitExtensionsForm
         private readonly EventHandler<GitModuleEventArgs> _gitModuleChanged;
 
         public FormInit(string dir, EventHandler<GitModuleEventArgs> gitModuleChanged)
+            : base(null, true)
```
Comment (mstv) [inline GitUI/CommandsDialogs/FormInit.cs:29]:
```suggestion
            : base(commands: null, enablePositionRestore: true)
```

**Comment by RussKie** [comment]
@NikolayXHD I've compared colours in Windows TaskDialog implementation, and it uses slightly different colours to what defined by the .NET runtime. Specifically the shades of the bottom panel:
image
(left a TaskDialog, right .NET Fx)

The native control uses #f0f0f0, and .NET Fx `ControlLight` is #e3e3e3. I'd like to define a custom colour via theme, but I'm a little stumped on how to best achieve it. Do I make it a part of `AppColor` set?

**Comment by RussKie** [comment]
> Is the layout with PuTTY considered?

Yep
image

**Comment by RussKie** [comment]
`FormStatus` change:

ge-status-dark
ge-status-light

**Comment by mstv** [comment]
Rebase, please, in order to resolve conflicts and for easier review.

**Comment by RussKie** [comment]
I'd like to have this merged asapp, I'm cleaning the `FormStatus` API (it is mind boggling spaghetti), and it depends on this.

File: GitUI/HelperDialogs/FormProcess.cs [hunk 2]
```diff
@@ -49,14 +49,14 @@ private FormProcess(ConsoleOutputControl outputControl, [CanBeNull] string proce
             ConsoleOutput.DataReceived += DataReceivedCore;
         }
 
-        public FormProcess(string process, ArgumentString arguments, string workingDirectory, string input, bool useDialogSettings)
-            : this(null, process, arguments, workingDirectory, input, useDialogSettings)
+        public FormProcess(GitUICommands commands, string process, ArgumentString arguments, string workingDirectory, string input, bool useDialogSettings)
+            : this(commands, null, process, arguments, workingDirectory, input, useDialogSettings)
```
Comment (mstv) [inline GitUI/HelperDialogs/FormProcess.cs:53]:
```suggestion
            : this(commands, outputControl: null, process, arguments, workingDirectory, input, useDialogSettings)
```

File: GitUI/HelperDialogs/FormStatus.cs [hunk 3]
```diff
@@ -24,8 +24,8 @@ private protected FormStatus()
             InitializeComponent();
         }
 
-        public FormStatus([CanBeNull] ConsoleOutputControl consoleOutput, bool useDialogSettings)
-            : base(true)
+        public FormStatus(GitUICommands commands, [CanBeNull] ConsoleOutputControl consoleOutput, bool useDialogSettings)
+            : base(commands, true)
```
Comment (mstv) [inline GitUI/HelperDialogs/FormStatus.cs:28]:
```suggestion
            : base(commands, enablePositionRestore: true)
```

File: GitUI/HelperDialogs/FormStatus.cs [hunk 4]
```diff
@@ -44,11 +44,19 @@ public FormStatus([CanBeNull] ConsoleOutputControl consoleOutput, bool useDialog
...
+
+            Controls.SetChildIndex(ProgressBar, 1);
+            ProgressBar.Dock = DockStyle.Bottom;
+
             InitializeComplete();
         }
 
         public FormStatus(Action<FormStatus> process, string text)
-            : this(new EditboxBasedConsoleOutputControl(), true)
+            : this(null, new EditboxBasedConsoleOutputControl(), true)
```
Comment (mstv) [inline GitUI/HelperDialogs/FormStatus.cs:59]:
```suggestion
            : this(commands: null, new EditboxBasedConsoleOutputControl(), true)
```

File: GitUI/UserControls/RevisionGrid/RevisionGridControl.cs [hunk 5]
```diff
@@ -2604,7 +2604,7 @@ private void LaunchRebase(string command)
                 LatestSelectedRevision.FirstParentId?.ToString(),
                 interactive: true, preserveMerges: false, autosquash: false, autoStash: true);
 
-            using (var formProcess = new FormProcess(null, rebaseCmd, Module.WorkingDir, null, true))
+            using (var formProcess = new FormProcess(UICommands, null, rebaseCmd, Module.WorkingDir, null, true))
```
Comment (mstv) [inline GitUI/UserControls/RevisionGrid/RevisionGridControl.cs:2607]:
```suggestion
            using (var formProcess = new FormProcess(UICommands, outputControl: null, rebaseCmd, Module.WorkingDir, input: null, useDialogSettings: true))
```

**Pull request merged**